### PR TITLE
fix: fetch connectors list from npm

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Fetch official connectors list from npm, this will be more accurate, since we will download connector from npm too.

## Testing

image

### CODE REVIEW

File: packages/core/src/connectors/add-connectors.ts [hunk 1]
```diff
@@ -5,21 +5,20 @@ import path from 'path';
...
 const execPromise = promisify(exec);
+const npmConnectorFilter = '@logto/connector-';
 
 const fetchOfficialConnectorList = async () => {
-  // Will change to "logto-io/connectors" once the new repo is ready.
-  const directories = await got
-    .get('https://api.github.com/repos/logto-io/connectors/contents/packages')
-    .json<Array<{ name: string }>>();
+  const { stdout } = await execPromise(`npm search ${npmConnectorFilter} --json`);
+  const packages = z.object({ name: z.string() }).array().parse(JSON.parse(stdout));
```
Comment (darcyYe) [inline packages/core/src/connectors/add-connectors.ts:19]: nit: should we consider using safeParse and try catch for JSON.parse here?

**Review by darcyYe — APPROVED** [review]

**Review by gao-sun — APPROVED** [review]

File: packages/core/src/connectors/add-connectors.ts (hunk 1 above)
Comment (gao-sun) [reply packages/core/src/connectors/add-connectors.ts:19]: no need. this one cannot fail

Comment (gao-sun) [reply packages/core/src/connectors/add-connectors.ts:19]: i mean we don't allow it fails

Comment (wangsijie) [reply packages/core/src/connectors/add-connectors.ts:19]: good call, is nice to have some error handlers for our CLI, created an issue to track LOG-4101

**Pull request merged**